ksql-engine/src/main/java/io/confluent/ksql/codegen/SqlToJavaVisitor.java

```diff
@@ -225,28 +219,32 @@ private String formatQualifiedName(QualifiedName name) {
     }
 
     protected Pair<String, Schema> visitLongLiteral(LongLiteral node, Boolean unmangleNames) {
-      functionArguments.addArgumentType(Schema.Type.INT64);
       return new Pair<>("Long.parseLong(\"" + node.getValue() + "\")", Schema.INT64_SCHEMA);
     }
 
     @Override
     protected Pair<String, Schema> visitIntegerLiteral(final IntegerLiteral node,
                                                        final Boolean context) {
-      functionArguments.addArgumentType(Schema.Type.INT32);
       return new Pair<>("Integer.parseInt(\"" + node.getValue() + "\")", Schema.INT32_SCHEMA);
     }
 
     @Override
     protected Pair<String, Schema> visitFunctionCall(FunctionCall node, Boolean unmangleNames) {
       String functionName = node.getName().getSuffix();
       UdfFactory udfFactory = functionRegistry.getUdfFactory(functionName);
+      FunctionArguments functionArguments = new FunctionArguments();
```

Comment:
There is a use of `functionArguments` at the bottom of this method that has no corresponding being/endFunction - so I think it can be removed.

Reply:
Plus I think it could just be replace with a local List<Schema.Type>

Reply:
Changed it into a list.